cdptools/pipelines/minimal_event_data.py

```diff
@@ -0,0 +1,10 @@
+from datetime import datetime
+from typing import NamedTuple, Optional
+
+
+class MinimalEventData(NamedTuple):
+    body: str
+    event_datetime: datetime
+    video_uri: str
```

Comment:
Hm I see, I wasn't aware that there could be multiple sessions within the same event. How do we handle this currently?

Reply:
Also for point 2, you mean `staging-seattle` would primarily just hold city specific stuff (scrapers, etc.) and pull in all the generalized tools as needed?

Reply:
We currently don't handle them really at all except stating that they are the exact same event:

Morning session: https://councildataproject.github.io/seattle/#/events/c3d10348-5c4a-4228-8297-95174cd58233
Afternoon session: https://councildataproject.github.io/seattle/#/events/e80a6bb2-4036-44c5-9013-2d3663a07101

Check the details on both of them, there is no differences what-so-ever it's just transcript and video differences.

Reply:
@JacksonMaxfield Given your suggestion about multiple sessions, how would filtering events by date work? 
We can't just put the date (without info on starting time) because it's possible to have sessions on different days?

Reply:
Hmmmm I don't believe it is possible to have multiple sessions over multiple days. I think what I really meant was that sessions have a time and events have a date.

Reply:
Then `MinimalEventData` needs a date field?

Reply:
Yes. But really I would just have them all be datetimes. Where the EventDatetime is just the earliest session datetime. Which is why it probably isnt needed and be computed.

Reply:
Basically, we can compute the event datetime from the provided session datetimes.

Reply:
This is all operating under the assumption that we want to handle sessions. If we dont want to just yet then we dont need this.

Reply:
From a scraping perspective, how do we decide to group sessions to a single event? Do we just make an assumption that if a video url/webpage has the same body name and is within a similar time as others they are sessions of the same event? 

Or is there some data that actually links sessions together explicitly? I'm not that familiar with the seattle channel website, but for the 2 sessions you linked, I looked through the seattle channel html and didn't see anything that connects them

Reply:
Could there be a case where a body has 2 events in the same day, but they're about different topics so they aren't technically sessions?

Reply:
I think the questions you just posed are enough for me to agree that we should, for now, only handle `Events` not `Sessions`.

Reply:
In those two linked events, they also share the same minutes items. In the future, that something to explore in trying to link sessions. I think we would have to make some sort of inference (probably can't guarantee 100% correct?) from body name, date, minutes items.

Reply:
on the seattlechannel, they also link the session in the description with: `Watch the afternoon session.`

Reply:
We actually already do a text-diff compare to match seattle channel events to legistar events. If there are multiple events for the same body during the day we find the match by finding the least different minutes items list. Which is why I figure we could session stuff.

Reply:
Because we are tossing this on the scraper side I think it would be okay to have sessions because as To pointed out, the seattle specific scraper could infer sessions by `regex`'ing "Watch the *** session" or similar.

Reply:
Sounds good, I'll work on adding logic to the parser for handling sessions and update it in this PR